README.md

```diff
@@ -22,29 +22,19 @@ Note that this middleware is designed to be used for Leaf front-end websites onl
 
 # Usage
 
-To use the Leaf Error Middleware, just add the middleware to your `Config` and then to your `droplet.json` (make sure you `import LeafErrorMiddleware` at the top):
+To use the Leaf Error Middleware, just add the middleware to your `configure.swift` (make sure you `import LeafErrorMiddleware` at the top):
 
 ```swift
-let config = Config()
-config.addConfigurable(middleware: LeafErrorMiddleware.init, name: "leaf-error"))
-let drop = Droplet(config)
-```
-
-This replaces the default error middleware in Vapor, so ***do not*** include the standard `error` in your `droplet.json`.
-
-***Note:*** You should ensure you set the error middleware as the first middleware in your `droplet.json` to so all errors get caught (unless you are using something like [Vapor Security Headers](https://github.com/brokenhandsio/VaporSecurityHeaders/)):
+// Must set the preferred renderer:
+config.prefer(LeafRenderer.self, for: TemplateRenderer.self)
 
-```json
-{
-    ...
-    "middleware": [
-        "leaf-error",
-        ...
-    ],
-    ...
+services.register { worker in
+    return try LeafErrorMiddleware(environment: worker.environment, log: worker.make())
```

Comment:
I'm wondering if we can get the `log` from the request. Environment is immutable so that shoudn't cause any issues but if two threads hit the logger at the same time could cause a crash

Reply:
Additionally the middleware should probably be a singleton (separate instance for each worker) which means it can be registered like the ErrorMiddleware

Reply:
https://github.com/vapor/vapor/blob/master/Sources/Vapor/Services/Services%2BDefault.swift#L69

ErrorMW retrieves the logger from the initializer though. Want me to still initialize it inside respond(...) ?

Reply:
Hmm actually it might be ok. I _think_ that that way will create a new instance for each worker, so there should be no access issues 👍